### ISSUE TITLE
Is it possible to translate the UI text? and to add Thumbsup inside a Jekyll page?

### ISSUE THREAD

**Issue by tunganh03** [thumbsup/thumbsup#176]
## Feature request

First of all, I love this gallery generator. But I would like to know how can I edit the English text that showed on the front page. I know that I can edit the text in the title, album name and footer but **how about the "Home", day of the week and months?** Because I am Vietnamese and I would like to translate English to Vietnamese.

![Ảnh chụp Màn hình 2019-06-26 lúc 00 16 46](https://user-images.githubusercontent.com/17230355/60119139-f0fdac80-97a7-11e9-9681-d65c39e4accb.png)

And there is one more thing I would like to ask:
**Is it possible to add Thumbsup inside a Jekyll page?**

That's all. Thank you for this awesome product.

**Comment by rprieto** [comment]
Hi, I'm glad you're enjoying thumbsup! Localisation is a very good idea.

I'll try to work on it soon. I'll add some notes for now of what's required:

- the dates are handled by the Moment library. We'd need a setting like `--locale vi_VN` so they're displayed the right way
- the other words (`Home`, `photos`) are currently hardcoded. Some options would be:
  - bundle translations for many languages, and load the right one based on the `--locale` setting above. This is the simplest to use, but requires thumbsup to "know" about every language.
  - have another setting like `--locale-file ...` so users can customise all text as they want. It could be a good starting point.

Il not sure about integrating with Jekyll. The simplest might be to merge the 2 folders after running them separately.

**Comment by rprieto** [comment]
Hi @tunganh03, version `2.12.0` is now out with the following settings:

- `--home-album-name <name>` to control the top-level album name
- `--locale <code>` to change the way dates are rendered, e.g. `--locale vi`

The only text left is "5 photos, 2 videos" which will be removed from thumbsup core to be part of the themes. This will take a little longer.

**Comment by zibous** [comment]
i use this helper method helpers/translate.js for the translation:

```
"use strict";
module.exports = (config, text) => {
    /*
     see config.json
     ...
     "translate": {
        "photos": "Fotos",
        "photo": "Foto",
        "albums": "Alben",
        "album": "FoAlbumto",
        "videos": "Filme",
        "video": "Film"
    }
    ...
    <div class="summary">{{translate @root.settings summary}}</div>
    ...
    */
    try {
        Object.entries(config.translate).forEach(([key, value]) => {
            text = text.replace(key, value)
        });
    } catch (e) {}
    return text
}
```

**Comment by zibous** [comment]
@rprieto 

Please can you also add the numberFormat based on the config.locale:

![Bildschirmfoto 2019-07-27 um 16 28 01](https://user-images.githubusercontent.com/30198737/61995765-06a71000-b08d-11e9-9f93-2c3624bc062a.png)


```
function itemCount(count, type) {
    if (count === 0) return ''
    var plural = (count > 1) ? 's' : ''
    return '' + count + ' ' + type + plural  
}
```

**Comment by rprieto** [comment]
Good idea about numberFormat, would it format the number like `1000`, `1,000`, `1 000` etc?

For the stats (e.g. `3 albums, 100 photos`), because it's the only text left I was thinking of:
- removing the `album.stats` string from the model
- making the themes responsible for it - this way they can display whatever they want

I do like your idea of a default text that's translated when available. I was looking at adding support for an external translation file using something like https://www.npmjs.com/package/node-polyglot or https://github.com/mashpie/i18n-node to handle pluralisation in different languages (e.g. languages with N plural forms).

**Comment by zibous** [comment]
@rprieto 

I try console.log(new Intl.NumberFormat('de-DE').format(number)); but it do not work.
.. to handle pluralisation in different languages sounds good 👍 

Thanks.
